changelog.d/8837.bugfix

```diff
@@ -0,0 +1 @@
+Have Synapse's register admin endpoint (/_synapse/admin/v1/register) return a 400 error on a missing mac.
```

Comment:
Something like the following might be clearer?

```suggestion
Fix a long standing bug in the register admin endpoint (`/_synapse/admin/v1/register`) when the `mac` field was not provided. The endpoint now properly returns a 400 error.
```

You can add "Contributed by <your name>." at the end if you'd like!